:book: add beta release issue template to the release tasks docs (#8981)
kubernetes-sigs/cluster-api: docs/release/release-tasks.md

 - Packet: https://github.com/kubernetes-sigs/cluster-api-provider-packet/issues/new
 - vSphere: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/new

> 

Reviewer: Q: Should we start putting stuff like this in a separate file? We could have docs/release/communication/file and then just link it from here

Reviewer 2: IMO it's fine to have here for now as a first step. This process should ideally be automated so it's not a load of copy-paste work though.

Author: Keeping it as is for now :)

> Below mentioned template can be used to update CAPI providers for the beta releases. The template was used to update the CAPI providers after `v1.5.0-beta.0` was released:

Reviewer 2: ```suggestion
Below mentioned template can be used to update CAPI providers for the beta releases. The versions and dates should be updated.
```

```
<!-- title -->

> # CAPI v1.5.0-beta.0 has been released and is ready for testing 

Reviewer 2: ```suggestion
# CAPI $RELEASE_TAG has been released and is ready for testing 
```

<!-- title -->

<!-- body -->

> CAPI `v1.5.0-beta.0` has been released and is ready for testing. 

Reviewer 2: ```suggestion

```

> Looking forward to your feedback before CAPI 1.5.0 release, the 25th July 2023!

Reviewer 2: ```suggestion
Looking forward to your feedback before the release on $FULL_RELEASE_DATE!
```

## For quick reference: 

> - [CAPI v1.5.0-beta.0 release notes](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.5.0-beta.0)

Reviewer 2: ```suggestion
- [CAPI $RELEASE_TAG release notes](https://github.com/kubernetes-sigs/cluster-api/releases/tag/$RELEASE_TAG)
```

- [Shortcut to CAPI git issues](https://github.com/kubernetes-sigs/cluster-api/issues)

## Following are the planned dates for the upcoming releases:
|Release|Expected Date|
|-----|-----|
|v1.5.0-beta.x | Tuesday 5th July 2023|
|release-1.5 branch created (Begin [Code Freeze])|Tuesday 11th July 2023|
|v1.5.0-rc.0 released|Tuesday 11th July 2023|
| release-1.5 jobs created | Tuesday 11th July 2023 |
| v1.5.0-rc.x released | Tuesday 18th July 2023 |

> | v1.5.0 released	| Tuesday 25th July 2023 |

Reviewer 2: TODO: Insert release calendar from `docs/release/releases/release-XXXX`

Author: Updated via aa868c3690288209fa0366854b483e85b2945093

> |-----|-----|
> |v1.5.0-beta.x | Tuesday 5th July 2023|
> |release-1.5 branch created (Begin [Code Freeze])|Tuesday 11th July 2023|
> |v1.5.0-rc.0 released|Tuesday 11th July 2023|
> | release-1.5 jobs created | Tuesday 11th July 2023 |
> | v1.5.0-rc.x released | Tuesday 18th July 2023 |
> | v1.5.0 released	| Tuesday 25th July 2023 |

Reviewer 2: ```suggestion
TODO: Insert updated release calendar from `docs/release/releases/release-XXXX`

|-----|-----|
|v1.5.0-beta.x | Tuesday 5th July 2023|
|release-1.5 branch created (Begin [Code Freeze])|Tuesday 11th July 2023|
|v1.5.0-rc.0 released|Tuesday 11th July 2023|
| release-1.5 jobs created | Tuesday 11th July 2023 |
| v1.5.0-rc.x released | Tuesday 18th July 2023 |
| v1.5.0 released	| Tuesday 25th July 2023 |
```

Author: Updated via aa868c3690288209fa0366854b483e85b2945093
